__tests__/mark_utxos_selected_as_input.test.js

```diff
@@ -0,0 +1,179 @@
+import { Storage, Transaction, Input, Output } from '@hathor/wallet-lib';
+import TestUtils from './test-utils';
+
+const walletId = 'stub_mark_inputs_as_used';
+
+function createCustomTxHex() {
+  const txId0 = '5db0a8c77f818c51cb107532fc1a36785adfa700d81d973fd1f23438b2f3dd74';
+  const txId1 = 'fb2fbe0385bc0bc8e9a255a8d530f7b3bdcebcd5ccdae5e154e6c3d57cbcd143';
+  const txId2 = '11835fae291c60fc58314c61d27dc644b9e029c363bbe458039b2b0186144275';
+  const tx = new Transaction(
+    [new Input(txId0, 0), new Input(txId1, 1), new Input(txId2, 2)],
+    [new Output(100, Buffer.from('0463616665ac', 'hex'))],
+    {
+      timestamp: 123,
+      parents: ['f6c83e3641a08ec21aebc01296ff12f5a46780f0fbadb1c8101309123b95d2c6'],
+    },
+  );
+
+  return tx.toHex();
+}
+
+describe('mark utxos selected_as_input api', () => {
+  let selectSpy;
+  const txHex = createCustomTxHex();
+
+  beforeAll(async () => {
+    selectSpy = jest.spyOn(Storage.prototype, 'utxoSelectAsInput');
```

Comment:
[question] Is it necessary to target the `prototype` or are you being extra explicit?

Reply:
It is necessary because in jest, `jest.spyOn(Storage, 'utxoSelectAsInput')` would only work if `utxoSelectAsInput` was a static method.
Since it is a normal method this mock would not affect the instances of `Storage`.
To affect the instances of `Storage` we need to mock the `Storage.prototype.utxoSelectAsInput`.

Reply:
Interesting! Thank you for the knowledge.